docs/how-to-add-new-output.md

```diff
@@ -0,0 +1,67 @@
+# How to add a new output type
+
+## Understanding the API
+
+The ClusterLogForwarder API is defined by a collection of Go struct types.
+
+[type ClusterLogForwarderSpec struct][cluster_log_forwarder] specifies
+* inputs to select logs
+* outputs to forward logs to remote targets
+* pipelines to connect inputs to outputs
+
+It supports an open-ended set of "output types", such as Loki, Kafka, Elasticsearch and so on.
+
+[type ClusterLogForwarderOutputSpec struct][cluster_log_forwarder] specifies common Name, Type, URL, Secret fields used by all outputs. The Secret fields references a K8s Secret object used to store any sensitive authentication or authorization data needed by the output.
+
+[type OutputTypeSpec struct][output_types] MAY add extra fields that are unique to the new output type.
```

Comment:
in addition we can say :
define a golang struct for the new type, and add it to the `OutputTypeSpec`